### PR TITLE
upgrade to latest build rules

### PR DESCRIPTION
**Pull request by mikedanese**
cc @mtaufen

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 git_repository(
     name = "io_bazel_rules_go",
-    commit = "d0142854a22a0dd98306280e897e64086289a0de",
```
Comment (ixdy) [inline WORKSPACE]: this bumps bazel up to golang 1.7.5, while our regular builds are still using 1.7.4.

Comment (mikedanese) [inline WORKSPACE]: I'm not terribly concerned about this yet considering we aren't using bazel build artificats much. We should probably do the pinning in our repo eventually.

**Comment by mikedanese** [comment]
@smarterclayton @thockin @wojtek-t @eparis can someone approve this change?

**Comment by mikedanese** [comment]
@k8s-bot bazel test this

**Pull request merged**